### DISCUSSION TITLE
file structure does not work on windows / synology dsm

### DISCUSSION THREAD

**Discussion by JyFu-py** [steilerDev/icloud-photos-sync#161] [Q&A] [answered]
Hello, 

After many days of sync (my library is huge) the file structure is finaly here, and it work like a charm on linux !!

but in windows the symbolic links does not work :'(
(the directory hash like .2762DC2D-42A2-4A3A-9F4F-5C8486A2BB49 is browsable)

and on synology dsm, the symbolic links is not here, only the hashed directory 

![image](https://user-images.githubusercontent.com/67822884/212439314-604b5d03-9a2f-4269-bcb7-3bc9361a9fcf.png)

thanks !

**Comment by steilerDev** [comment]
This seems to be an issue with Synology - maybe check if this is a known issue on their end?

Using [dperson/samba](https://hub.docker.com/r/dperson/samba/) as fileshare, accessing the folder structure on Windows 10 works fine on my side. Same holds true for MacOS Ventura and the Files.app on  iOS 16.2.

**Reply by steilerDev** [reply]
Maybe down the line, I'll adjust the file structure to have a metadata file for the folder UUID instead of linking - but this would require some serious rework that I won't be able to do any time soon.

**Comment by JyFu-py** [comment, marked as answer]
ok the problem is samba,

the option "follow simlink" must be activated

![image](https://user-images.githubusercontent.com/67822884/212499637-dc10d23b-af1b-4d4b-8c63-0c8e2da373f8.png)

It works perfectly now !

**Comment by lonevvolf** [comment]
On a side note - it seems that the Synology File Station app doesn't show symbolic links, even if they are there and working.  If you mount the same share via SMB, they show up as expected.
